pkg/controllers/report/background/controller.go

```diff
@@ -123,6 +123,9 @@ func (c *controller) Run(ctx context.Context, workers int) {
 }
 
 func (c *controller) addPolicy(obj kyvernov1.PolicyInterface) {
+	if !utils.CanBackgroundProcess(logger, obj) {
```

Comment:
You probably don't want to check this here (and in other handlers)

Reply:
It would reduce some work if return earlier, no?

Reply:
We should let reconciliation happen, especially in update/delete handlers.

Reply:
If a policy changes and background becomes false, we must reconcile reports for example.

Reply:
Makes sense! Updating the PR.